dashboard/src/utils/transposeCSV.ts

```diff
@@ -0,0 +1,14 @@
+import csvToJson from "csvtojson";
+
+export default async function(csvString: string) {
+  /* Parses a csv file into an object mapping csv headers to column data for
+   * easier creation of partial csv files.
+   */
+  const csvJson = await csvToJson({ checkColumn: true }).fromString(csvString);
```

Comment:
have you looked at papaparse https://www.npmjs.com/package/papaparse? seems like it may be smaller w/ no dependencies and just as fast. Using the preview option would likely be helpful for getting the headers and the first few lines to display, and then finish parsing only when actually posting the data. The metadata/errors during parsing might be helpful, and you can probably get rid of `mapToCSV` using `unparse` and specifying the columns you want to keep.

Reply:
Ah, yes. I had looked at it but only when considering building the arrow tables. But now that this is using purely csv papaparse seems like a better alternative.  Thanks.